components/infobox/wikis/starcraft/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,510 @@
+	}
+
+	local processPlacement = function(placement)
+		--handle earnings
+		earnings, earningsTotal = CustomPlayer._addPlacementToEarnings(earnings, earningsTotal, placement)
+
+		--handle medals
+		medals = CustomPlayer._addPlacementToMedals(medals, placement)
+	end
+
+	Lpdb.executeMassQuery('placement', queryParameters, processPlacement)
+
+	if #_achievements > 0 then
+		Variables.varDefine('achievements', Json.stringify(_achievements))
+	end
+	if #_awardAchievements > 0 then
+		Variables.varDefine('awardAchievements', Json.stringify(_awardAchievements))
+	end
+	CustomPlayer._setVarsFromTable(earnings)
+	CustomPlayer._setVarsFromTable(medals)
+
+	return earningsTotal, earnings['total']
+end
+
+function CustomPlayer._addPlacementToEarnings(earnings, earningsTotal, data)
+	local mode = _EARNING_MODES[(data.players or {}).type or ''] or 'other'
+	if not earnings[mode] then
+		earnings[mode] = {}
+	end
+	local year = string.sub(data.date, 1, 4)
+	earnings[mode][year] = (earnings[mode][year] or 0) + data.individualprizemoney
+	earnings['total'][year] = (earnings['total'][year] or 0) + data.individualprizemoney
+	earningsTotal = (earningsTotal or 0) + data.individualprizemoney
+
+	return earnings, earningsTotal
+end
+
+function CustomPlayer._addPlacementToMedals(medals, data)
+	if data.liquipediatiertype ~= 'Qualifier' then
+		local place = CustomPlayer._getPlacement(data.placement)
+		CustomPlayer._setAchievements(data, place)
+		if
+			(data.players or {}).type == 'solo'
+			and place and place <= 3
+		then
+			local tier = data.liquipediatier or 'undefined'
+			if not medals[place] then
+				medals[place] = {}
+			end
+			medals[place][tier] = (medals[place][tier] or 0) + 1
+			medals[place]['total'] = (medals[place]['total'] or 0) + 1
+			medals['total'][tier] = (medals['total'][tier] or 0) + 1
+		end
+	end
+
+	return medals
+end
+
+function CustomPlayer._setVarsFromTable(table)
+	for key1, item1 in pairs(table) do
+		for key2, item2 in pairs(item1) do
+			Variables.varDefine(key1 .. '_' .. key2, item2)
+		end
+	end
+end
+
+function CustomPlayer._getPlacement(value)
+	if String.isNotEmpty(value) then
+		value = mw.text.split(value, '-')[1]
+		if Table.includes(_ALLOWED_PLACES, value) then
+			return tonumber(value)
+		end
+	end
+end
+
+function CustomPlayer._setAchievements(data, place)
+	local tier = tonumber(data.liquipediatier)
+	if CustomPlayer._isAwardAchievement(data, tier) then
+		table.insert(_awardAchievements, data)
+	elseif #_achievements < _NUMBER_OF_ALLOWED_ACHIEVEMENTS then
+		table.insert(_achievements, data)
+	end
+end
+
+function CustomPlayer._isAwardAchievement(data, tier)
+	return String.isNotEmpty((data.extradata or {}).award) and (
+		tier == 1 or
+		tier == 2 and data.individualprizemoney > 50
+	)
+end
+
+function CustomPlayer:getWikiCategories(categories)
+	if _args.tlpdsospa then
+		table.insert(categories, 'SOSPA Players')
+	end
+
+	if _lpdbData.nationality ~= _KOREAN then
```

Comment:
Can this be done without looking at LPDB Data?

Reply:
yeah can just use `_args.country` as it gets standardized into that in the commons already